bus-mapping/src/external_tracer.rs

```diff
@@ -1,9 +1,86 @@
 //! This module generates traces by connecting to an external tracer
-use crate::eth_types::{self, Address, GethExecStep, Word};
-use crate::BlockConstants;
+use crate::eth_types::{self, Address, Block, GethExecStep, Hash, Word, U64};
 use crate::Error;
 use geth_utils;
 use serde::Serialize;
+use std::str::FromStr;
+
+/// Definition of all of the constants related to an Ethereum block and
+/// chain to be used as setup for the external tracer.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub struct BlockConstants {
+    hash: Hash, // Until we know how to deal with it
+    coinbase: Address,
+    timestamp: Word,
+    number: U64, // u64
+    difficulty: Word,
+    gas_limit: Word,
+    chain_id: Word,
+    base_fee: Word,
+}
+
+impl BlockConstants {
+    /// Generate a BlockConstants from an ethereum block, useful for testing.
+    pub fn from_eth_block<TX>(
+        block: &Block<TX>,
+        chain_id: &Word,
+        &coinbase: &Address,
+    ) -> Self {
+        Self {
+            hash: block.hash.unwrap(),
+            coinbase,
+            timestamp: block.timestamp,
+            number: block.number.unwrap(),
+            difficulty: block.difficulty,
+            gas_limit: block.gas_limit,
+            chain_id: *chain_id,
+            base_fee: block.base_fee_per_gas.unwrap(),
+        }
+    }
+
+    /// Generate a new mock BlockConstants, useful for testing.
+    pub fn mock() -> Self {
```

Comment:
Should this be `#cfg[(test)]`?

Reply:
I noticed this function is not used (in favour of `from_eth_block` so I'll remove it!